src/element.ts

```diff
@@ -0,0 +1,275 @@
+// [VexFlow](http://vexflow.com) - Copyright (c) Mohit Muthanna 2010.
+// @author Mohit Cheppudira
+//
+// ## Description
+//
+// This file implements a generic base class for VexFlow, with implementations
+// of general functions and properties that can be inherited by all VexFlow elements.
+
+import { Vex } from './vex';
+import { Registry } from './registry';
+import { Flow } from './tables';
+import { BoundingBox } from './boundingbox';
+import { Font } from './smufl';
+
+/** Element attributes. */
+export interface ElementAttributes {
+  [name: string]: string;
+  id: string;
+  type: string;
+}
+
+/** Contexts common interface */
+export interface RenderContext {
```

Comment:
Maybe a separate `rendercontext.ts` for this.

Reply:
I will move it to types/common if you agree.

Reply:
Okay, that's fine for now. I want to be careful not to bloat `types/common`.

This might be a good opportunity to add some directory structure. Create a `renderer` directory with it's own `.d.ts` file and move all the rendering code there: `svgcontext`, `canvascontext`, `raphaelcontext`, and `renderer.js`.

Maybe in the next PR.